openmoc/materialize.py

```diff
@@ -138,7 +141,11 @@ def load_from_hdf5(filename='mgxs.h5', directory='mgxs',
                     if isinstance(domain_spec, int):
                         material = openmoc.Material(id=domain_spec)
                     else:
-                        material = openmoc.Material(name=domain_spec)
+                        # Reproducibly hash the domain name into an integer ID
+                        domain_id = \
+                            int(hashlib.md5(domain_spec).hexdigest()[:4], 16)
```

Comment:
This simply specifies the how many decimal places to include in the hashed ID. I'll be honest that the method *may* allow for hash collisions and redundant material IDs...